### PR TITLE
feat: dummy pricing provider

### PR DESCRIPTION
**Pull request by 0xnigir1**
# 🤖 Linear

Closes GIT-136 GIT-144

## Description
- DummyPricingProvider for testing, dev environments to avoid rate limiting
- write a Factory to instantiate a pricing provider from Env config
- makes `timestampEnd` optional for `IPricingProvider`

## Checklist before requesting a review

-   [x] I have conducted a self-review of my code.
-   [x] I have conducted a QA.
-   [x] If it is a core feature, I have included comprehensive tests.

### CODE REVIEW

File: apps/processing/src/config/env.ts [hunk 1]
```diff
@@ -23,13 +23,38 @@ const validationSchema = z.object({
...
+        }
+
+        return {
+            pricingSource: val.PRICING_SOURCE,
+            apiKey: val.COINGECKO_API_KEY,
+            apiType: val.COINGECKO_API_TYPE,
+            ...val,
+        };
+    });
+
```
Comment (0xkenj1) [inline apps/processing/src/config/env.ts:57]: We can move this into a different file `in the future`(not now), to have a more readable and modular code.

**Review by 0xkenj1 — APPROVED** [review]
approved with smoll comment :)

File: apps/processing/.env.example [hunk 2]
```diff
@@ -13,5 +13,7 @@ INDEXER_ADMIN_SECRET=testing
 
 IPFS_GATEWAYS_URL=["https://ipfs.io","https://gateway.pinata.cloud","https://dweb.link", "https://ipfs.eth.aragon.network"]
 
+PRICING_SOURCE= #coingecko | dummy
```
Comment (0xyaco) [inline apps/processing/.env.example:16]: Is that `#coingecko` ok?

File: packages/pricing/src/factory/index.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        let pricingProvider: IPricingProvider;
+
+        switch (options.pricingSource) {
+            case "dummy":
+                pricingProvider = new DummyPricingProvider(options.dummyPrice);
+                break;
+            case "coingecko":
+                if (!deps?.logger) {
+                    throw new MissingDependencies();
+                }
```
Comment (0xyaco) [inline packages/pricing/src/factory/index.ts:40]: I'm wondering if we should standardize the dependencies validation with some kind of generic pattern (with the help of zod maybe?) while initializing objects.

File: packages/pricing/src/providers/dummy.provider.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+/**
+ * DummyPricingProvider class that implements the IPricingProvider interface.
+ * This provider returns a configurable fixed price (defaults to 1) for any token code.
+ * Used primarily for testing purposes when actual token prices are not needed.
+ */
+export class DummyPricingProvider implements IPricingProvider {
+    constructor(private readonly dummyPrice: number = 1) {}
+
+    /* @inheritdoc */
+    async getTokenPrice(
```
Comment (0xyaco) [inline packages/pricing/src/providers/dummy.provider.ts:14]: Extremely minor detail but the `async` keyword here is not necessary given the `Promise.resolve()` usage

**Review by 0xyaco — APPROVED** [review]
Super minor comments/questions, good to go!

File: apps/processing/.env.example (hunk 2 above)
Comment (0xnigir1) [reply apps/processing/.env.example:16]: is # for comment, in next PR i'll add a whitespace

File: packages/pricing/src/providers/dummy.provider.ts (hunk 4 above)
Comment (0xnigir1) [reply packages/pricing/src/providers/dummy.provider.ts:14]: ooo didn't know that

File: packages/pricing/src/factory/index.ts (hunk 3 above)
Comment (0xnigir1) [reply packages/pricing/src/factory/index.ts:40]: you mean inside constructors or having a zod schema here instead of ` if (!deps?.logger)`

Comment (0xyaco) [reply packages/pricing/src/factory/index.ts:40]: Zod schema could be a solution yeah. I'm thinking in general, whenever a service needs dependencies and those dependencies might need to accomodate themselves to some particular schema (like optional dependencies, some dependencies need to be present based on some other dependency value, etc)

I guess that applying a zod schema could cover 99% of the cases, I'll apply it in my code and I'll let you know how happy/sad it is to use it lol

**Pull request merged**